controllers/argocd/keycloak.go

```diff
@@ -1090,8 +1090,19 @@ func (r *ReconcileArgoCD) updateArgoCDConfiguration(cr *argoproj.ArgoCD, kRouteU
 		log.Error(err, message)
 		return fmt.Errorf("%s error: %w", message, err)
 	}
-
-	argoRBACCM.Data["scopes"] = "[groups,email]"
+	if cr.Spec.RBAC.Scopes != nil && *cr.Spec.RBAC.Scopes != "" {
+		argoRBACCM.Data["scopes"] = *cr.Spec.RBAC.Scopes
+	} else {
+		argoRBACCM.Data["scopes"] = "[groups,email]"
+		scopes := argoRBACCM.Data["scopes"]
+		cr.Spec.RBAC.Scopes = &scopes
+		err = r.Client.Update(context.TODO(), cr) // Update the Spec.RBAC.Scopes with the default scopes added by keycloak.
```

Comment:
I think we should handle this in code instead of updating the rbac field in ArgoCD CR. I feel operator should respect `spec.rbac` when configuring keycloak. If the value is set, operator shouldn't update the configmap with keycloak related this. But if the `[groups,email]` is necessary for keycloak, then we have no option but to override the configmap field with keycloak related things. In this case, we should log it properly in the operator logs and handle it accordingly in `.spec.rbac` logic to avoid conflicting write.